scripts/validate-chart-versions.sh

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+GIT_REPO_ROOT=$(git rev-parse --show-toplevel)
+BUILD_DIR="${GIT_REPO_ROOT}/chartbuild"
+TOOLS_DIR="${BUILD_DIR}/tools"
+CHARTS="${GIT_REPO_ROOT}/deploy/charts"
+export PATH="${TOOLS_DIR}:${PATH}"
+
+EXIT_CODE=0
+if $(git describe HEAD --tags | grep -Eq "^v[0-9]+(\.[0-9]+)*(-[a-z0-9]+)?$"); then
+    LAST_RELEASE_HASH=$(git rev-list --tags --max-count=1 --skip=1 --no-walk)
+else 
+    TAG=$(git describe HEAD --tags | grep -Eo "^v[0-9]+(\.[0-9]+)*")
+    LAST_RELEASE_HASH=$(git rev-list -1 $TAG)
+fi
+LAST_RELEASE_TAG=$(git describe $LAST_RELEASE_HASH --tags)
```

Comment:
I would find a comment on this block helpful since it took me a bit to grok. Writing down my understanding, please correct me or feel free to copy it in:

```
# Determine the commit hash and tag of the previous release:
#
#     * If the repository head is at a release tag (e.g. v1.2.3-rc4), select
#       the release before that (e.g. v1.2.3-rc3).
#     * If the repository head has advanced from a release tag (e.g. v1.0-g0123abcd),
#       select the release the current head is based on (e.g. v1.0).
```